File: FrontEnd/assets/js/Domicilio.js
function Domicilio(tipo, rfc, nombre,calle,colonia,municipio, estado, pais,codigoPostal,noExterior){
    this.tipo = this.IsNullOrEmpty(tipo);  
    this.rfc = this.IsNullOrEmpty(rfc);
    this.nombre = this.IsNullOrEmpty(rfc);   
    this.calle = this.IsNullOrEmpty(calle);
    this.colonia = this.IsNullOrEmpty(colonia);
    this.municipio = this.IsNullOrEmpty(municipio);
    this.estado = this.IsNullOrEmpty(estado);
    this.pais = this.IsNullOrEmpty(estado);
    this.codigoPostal = this.IsNullOrEmpty(codigoPostal);    
    this.noExterior = this.IsNullOrEmpty(noExterior);
    return this;
}

Domicilio.prototype.IsNullOrEmpty= function(param){
    if (typeof(param) === 'undefined' || param === null ) {
        return '';
    }else {
        return param;
    }
}

File: App/Models/Models/Domicilio.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Models
{
    public class Domicilio
    {
        public string calle { get; set; }
        public string colonia { get; set; }
        public string municipio { get; set; }
        public string estado { get; set; }
        public string pais { get; set; }
        public string codigoPostal { get; set; }
        public string noExterior { get; set; }
    }
}
